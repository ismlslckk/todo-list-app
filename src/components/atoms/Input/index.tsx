const Input = (props: any) => {

    return (

        <></>
    );

};

export default Input;