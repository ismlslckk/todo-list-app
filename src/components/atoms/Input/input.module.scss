.baseInput{
    min-width: 300px;
}