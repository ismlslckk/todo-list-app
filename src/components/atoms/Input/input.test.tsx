import { render } from '@testing-library/react';
import reactTestRenderer from 'react-test-renderer';
import Input from '@/components/atoms/Input';

describe('input component tests', () => {
  test('expect input should get className from props', async () => {
    const input:any = render(<Input className="form-control" />);

    expect(input.container.firstChild.classList.contains('form-control')).toBe(true);
  });

  test('expect input should be readOnly', async () => {
    const input:any = render(<Input className="form-control" readOnly />);

    expect(input.container.firstChild).toHaveProperty('readOnly');
  });

  test('expect input should be disable', async () => {
    const input:any = render(<Input className="form-control" disabled />);

    expect(input.container.firstChild).toHaveProperty('disabled');
  });

  it('renders correctly', () => {
    const tree = reactTestRenderer
      .create(<Input />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
