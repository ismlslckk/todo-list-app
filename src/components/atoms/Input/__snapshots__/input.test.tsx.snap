// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component tests renders correctly 1`] = `
<input
  className="form-control"
/>
`;
